packages/graphback/src/database/initialization/strategies/CreateDatabaseIfNotExists.ts

```diff
@@ -0,0 +1,24 @@
+import { InputModelTypeContext } from '../../../input/ContextTypes';
+import { DatabaseConnectionOptions } from '../../DatabaseConnectionOptions';
+import { DatabaseContextProvider } from '../../migrations/DatabaseContextProvider';
+import { DatabaseSchemaManager } from '../../migrations/DataResourcesManager';
+import { DatabaseInitializationStrategy } from '../DatabaseInitializationStrategy';
+
+/**
+ * Database initialization strategy that will create a database on initialization - if one does not exist.
+ *
+ * @export
+ * @class CreateDatabaseIfNotExists
+ * @implements {DatabaseInitializationStrategy}
+ */
+export class CreateDatabaseIfNotExists implements DatabaseInitializationStrategy {
+  private schemaManager: DatabaseSchemaManager;
+  constructor(databaseOptions: DatabaseConnectionOptions) {
+    this.schemaManager = new DatabaseSchemaManager(databaseOptions.client, databaseOptions.connectionOptions);
+  }
+
+  public async init(context: DatabaseContextProvider, types: InputModelTypeContext[]): Promise<void> {
+    await this.schemaManager.createDatabaseResources(context, types);
+    await this.schemaManager.createDatabaseRelations(context, types);
```

Comment:
Relations are tricky as cascade rules and constraints might prefent things. 
For now this approach is ok, but later on we need to think how to properly manage relations.

Reply:
Fully agree. This was never intended to be the final solution but this PR needed some sort of scope limit.